core/driver/sources/src/main/scala/org/virtuslab/ideprobe/WaitLogic.scala

```diff
@@ -0,0 +1,29 @@
+package org.virtuslab.ideprobe
+
+import org.virtuslab.ideprobe.wait._
+import pureconfig.ConfigReader
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+
+trait WaitLogic {
+  def await(driver: ProbeDriver): Unit
+
+  def and(waitCondition: ProbeDriver => WaitDecision): WaitLogic
+
+  def or(waitCondition: ProbeDriver => WaitDecision): WaitLogic
+
+  def doWhileWaiting(maxFrequency: FiniteDuration)(code: => Unit): WaitLogic = {
+    val backoff = new ExecutionBackoff(maxFrequency)
+    or { _ =>
+      backoff.execute { code }
+      WaitDecision.Done
```

Comment:
I'm not sure if I understand it correctly. Imo we should either return `WaitDecision.Keep`, or call it with `and`. Otherwise it will always terminate until one check (because we're doing something like `or true`)

Reply:
WaitDecision answers the question `needToWait`. Perhaps I should invert this to be more intuitive.

Reply:
done